src/OpenTelemetry/Trace/ActivitySourceAdapter.cs

```diff
@@ -107,6 +111,14 @@ internal void UpdateProcessor(ActivityProcessor processor)
             this.activityProcessor = processor;
         }
 
+        private static Action<Activity, ActivityKind> CreateActivityKindSetter()
```

Comment:
Probably shouldn't have a function as this is not meant to be called multiple times?

Reply:
Just moving the current code as is from ActivityExtension to ActivitySourceAdapter in this PR. If we have scope for improving this, let me take that separately.